### ISSUE TITLE
Always 400 Bad Request with Nginx

### ISSUE THREAD

**Issue by MnM001** [Forceu/Gokapi#69]
Hi,

Testing with nginx (swag with letsencrypt from linuxserver.io) I always get 400 Bad Request.
Accessing via IP it works as it should but of course I cant share on the internet.

My setup is:
No SSL
Public facing URL - is my URL used by Nginx
Redirection is above URL/admin
Authentication is username and password
Storage - local storage + As above
Encryption - level 0

I have other apps behind nginx (like 20+ of them) - they all work fine.

I am not sure what the issues is. limited logs from docker (xxx are my changes):

██████   ██████  ██   ██  █████  ██████  ██ 
██       ██    ██ ██  ██  ██   ██ ██   ██ ██ 
██   ███ ██    ██ █████   ███████ ██████  ██ 
██    ██ ██    ██ ██  ██  ██   ██ ██      ██ 
 ██████   ██████  ██   ██ ██   ██ ██      ██ 
                                             
Gokapi v1.5.2 starting
Please open http://xxx.xxx.xxx.xxx:53842/setup to setup Gokapi.
Saving new files to local storage
Binding webserver to :53842
Webserver can be accessed at https://xxx.xxxxxx.xxx/admin
Press CTRL+C to stop Gokapi

**Comment by Forceu** [comment]
Thanks for your feedback. When exactly do you encounter the error? E.g. when uploading a file or when downloading one?

**Comment by MnM001** [comment]
Hi,

Whenever I brose to the URL - /admin or /login or just the base URL.
It never works at all - so after I finish my config via the IP address then I get redirected to the URL and I always get the 400 error.

**Comment by Forceu** [comment]
That sounds like your reverse proxy is not configured correctly. Can you please post the nginx site configuration?

**Comment by MnM001** [comment]
Yes sure - but it is configured correctly. I have 20+ other config files in Nginx working fine. 

## Version 2021/05/18

server {
    listen 443 ssl;
    #listen [::]:443 ssl;

    server_name gokapi.*;

    include /config/nginx/ssl.conf;
    include /config/nginx/errors.conf;

    client_max_body_size 500M;
    client_body_buffer_size 128k;
    disable_symlinks off;

    # enable for ldap auth, fill in ldap details in ldap.conf
    #include /config/nginx/ldap.conf;

    # enable for Authelia
    #include /config/nginx/authelia-server.conf;

    location / {
        # enable the next two lines for http auth
        #auth_basic "Restricted";
        #auth_basic_user_file /config/nginx/.htpasswd;

        # enable the next two lines for ldap auth
        #auth_request /auth;
        #error_page 401 =200 /ldaplogin;

        # enable for Authelia
        #include /config/nginx/authelia-location.conf;
        include /config/nginx/errors.conf;
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;

        proxy_buffering off;

        proxy_set_header X-Forwarded-Host $http_cf_connecting_ip;
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto http;

        set $upstream_app xxx.xxx.xxx.xxx;
        set $upstream_port 53842;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
    }
        if ( $scheme = http ) {
                return 301 https://$server_name$request_uri;
        }


        location ~ /\.ht {
                deny all;
        }
}

Similar config files for the other Nginx proxies - all working.

**Comment by Forceu** [comment]
Can you connect to the original ip and port from that machine? Even though I don't know whats included in the errors.conf, proxy.conf and resolver.conf it looks like it should be working. The setup worked without any problems? And are you running the app as a docker container? If yes please post the command you used for creating the container.

**Comment by MnM001** [comment]
Hi,

Yes I can connect to the app via IP address and port (xxx.xxx.xxx.xxx:53842). No problems at all connecting this way.
The gokapi setup via IP worked just fine.

The docker compose I use is:

    gokapi:
        image: f0rc3/gokapi:latest
        container_name: gokapi
        hostname: gokapi
        volumes:
            - /docker/containers/gokapi/config:/app/config
            - /docker/containers/gokapi/config/data:/app/data
        networks:
            - internal
        ports:
            - 53842:53842
        restart: unless-stopped

proxy.conf:

# Proxy Connection Settings
proxy_buffers 32 4k;
proxy_connect_timeout 240;
proxy_headers_hash_bucket_size 128;
proxy_headers_hash_max_size 1024;
proxy_http_version 1.1;
proxy_read_timeout 240;
proxy_redirect  http://  $scheme://;
proxy_send_timeout 240;

# Proxy Cache and Cookie Settings
proxy_cache_bypass $cookie_session;
#proxy_cookie_path / "/; Secure"; # enable at your own risk, may break certain apps
proxy_no_cache $cookie_session;

# Proxy Header Settings
proxy_set_header Connection $connection_upgrade;
proxy_set_header Early-Data $ssl_early_data;
proxy_set_header Host $host;
proxy_set_header Proxy "";
proxy_set_header Upgrade $http_upgrade;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Host $host;
proxy_set_header X-Forwarded-Proto https;
proxy_set_header X-Forwarded-Ssl on;
proxy_set_header X-Real-IP $remote_addr;



errors.conf:

error_page 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 421 422 423 424 426 428 429 431 451 500 501 502 503 504 505 506 507 508 510 511 /errors.html;

location = /errors.html {
           ssi on;
           internal;
           root /config/www/nginx-errors;
proxy_intercept_errors on;
}

location = /style.css {
    root /config/www/nginx-errors;
}




resolver.conf

resolver  127.0.0.11 valid=30s;

**Comment by Forceu** [comment]
The session cookie will probably be cached with this configuration, which might lead to problems, but I am not sure if that is the cause here.

Does it work if you uncomment the ` include /config/nginx/errors.conf `and `include /config/nginx/proxy.conf;` and reload the config? If yes, you could then find out with trial and error what configuration is causing the problem

**Comment by MnM001** [comment]
I have tried that just now - commented out the errors.conf  and proxy.conf. It still doesn't work but the erro I now get is:

404 page not found

**Comment by MnM001** [comment]
Commented out proxy.conf only = 404 page not found

**Comment by MnM001** [comment]
So strange... I made only 1 change and it got it going. Really not sure why as I reversed the change and it still works.

So I edited confgi.jason file and changed     "RedirectUrl": "https://xxxx.xxxxx.com/admin", to   "RedirectUrl": "https://gokapi.remotedpi.online/login",

restarted the docker container and it works fine with the URL.
Reversed the change to  "RedirectUrl": "https://xxxx.xxxxx.com/admin", and restarted the docker - it still works fine.

Strange...
